src/data-access/user.ts

```diff
@@ -0,0 +1,83 @@
+import { Op } from 'sequelize';
+import { UserModel, initUserModel } from '../models/user';
+
+import { User } from '../types/user';
+
+function convertToUser(model: UserModel): User | null {
+  if (!model) {
+    return null;
+  }
+
+  return {
+    id: model.id,
+    login: model.login,
+    password: model.password,
+    age: model.age,
+    isDeleted: model.isDeleted
+  };
+}
+
+export interface IAccessorToUserData<T> {
+  getById(id: string): Promise<T | null>;
+  createOrUpdate(user: User): Promise<User | null>;
```

Comment:
it is better to define separate methods for this operations

Reply:
done